From 78a641a3a6bd7aacc3558a39d561e0402f2221b8 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Sun, 16 Oct 2022 17:58:25 +0200
Subject: [PATCH] Preceise RT linking

---
 tests/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git tests/CMakeLists.txt tests/CMakeLists.txt
index 03af858..93d14c4 100644
--- tests/CMakeLists.txt
+++ tests/CMakeLists.txt
@@ -30,6 +30,7 @@
 #
 # This macro will create a binary from `NAME.cc' and add it to the test suite.
 #
+find_library(RT_LIB rt)
 macro(ADD_TESTCASE NAME GENERATED)
   if (${GENERATED} STREQUAL TRUE)
    add_unit_test(${NAME} ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.cc)
@@ -39,6 +40,11 @@ macro(ADD_TESTCASE NAME GENERATED)
 
   pkg_config_use_dependency(${NAME} tinyxml)
   target_link_libraries(${NAME} ${PROJECT_NAME})
+  if(RT_LIB)
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME} ${RT_LIB})
+  else()
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME} )
+endif()
 endmacro(ADD_TESTCASE)
 
 config_files(srdfparser.cc)
-- 
2.25.1
